apps/api/v2/src/ee/event-types/event-types_2024_06_14/services/input-event-types.service.ts

```diff
@@ -114,7 +227,245 @@ export class InputEventTypesService_2024_06_14 {
     return !!res ? res : {};
   }
 
+  transformInputBookerLayouts(inputBookerLayouts: CreateEventTypeInput_2024_06_14["bookerLayouts"]) {
+    return transformBookerLayoutsApiToInternal(inputBookerLayouts);
+  }
+
+  transformInputConfirmationPolicy(
+    requiresConfirmation: CreateEventTypeInput_2024_06_14["confirmationPolicy"]
+  ) {
+    return transformConfirmationPolicyApiToInternal(requiresConfirmation);
+  }
   transformInputRecurrignEvent(recurrence: CreateEventTypeInput_2024_06_14["recurrence"]) {
     return transformRecurrenceApiToInternal(recurrence);
   }
+
+  transformInputEventTypeColor(color: CreateEventTypeInput_2024_06_14["color"]) {
+    return transformEventColorsApiToInternal(color);
+  }
+
+  transformInputSeatOptions(seats: CreateEventTypeInput_2024_06_14["seats"]) {
+    return transformSeatsApiToInternal(seats);
+  }
+  async validateEventTypeInputs(
+    eventTypeId?: number,
+    seatsEnabled?: boolean,
+    locations?: ReturnType<typeof this.transformInputLocations>,
+    requiresConfirmation?: boolean,
+    eventName?: string
+  ) {
+    // Retrieve event type details from the database if eventTypeId is provided
+
+    if (eventTypeId != null) {
+      const eventTypeDb = await this.eventTypesRepository.getEventTypeWithSeats(eventTypeId);
+      const seatsEnabledDb = !!eventTypeDb?.seatsPerTimeSlot && eventTypeDb.seatsPerTimeSlot > 0;
+      const locationsDb = this.outputEventTypesService.transformLocations(eventTypeDb?.locations);
+      const requiresConfirmationDb = eventTypeDb?.requiresConfirmation ?? false;
+
+      seatsEnabled &&
+        (await this.validateSeatsInput(
+          locations,
+          requiresConfirmation,
+          locationsDb,
+          requiresConfirmationDb,
+          true
+        ));
+      locations &&
+        locations.length > 1 &&
+        (await this.validateLocationsInput(seatsEnabled, locations, seatsEnabledDb, true));
+      requiresConfirmation &&
+        (await this.validateRequiresConfirmationInput(
+          requiresConfirmation,
+          seatsEnabled,
+          seatsEnabledDb,
+          true
+        ));
+      eventName && (await this.validateCustomEventNameInput(eventName));
+    } else {
+      seatsEnabled &&
+        (await this.validateSeatsInput(locations, requiresConfirmation, undefined, undefined, false));
+      locations?.length > 1 && (await this.validateLocationsInput(seatsEnabled, locations, undefined, false));
+      requiresConfirmation &&
+        (await this.validateRequiresConfirmationInput(requiresConfirmation, seatsEnabled, undefined, false));
+      eventName && (await this.validateCustomEventNameInput(eventName));
+    }
+  }
+  async validateSeatsInput(
+    locations?: ReturnType<typeof this.transformInputLocations>,
+    requiresConfirmation?: boolean,
+    locationsDb?: ReturnType<typeof this.transformInputLocations>,
+    requiresConfirmationDb?: boolean,
+    validateDb = false
```

Comment:
If we update, why do we need information about the existing event type in database? Not saying it is not necessary i just dont understand why.

Just from checking code it is hard to understand the purpose of `validateDb` . Also, if a function like `validateSeatsInput` has optional boolean its an indicator that this function should be split in 2 .

Reply:
https://github.com/calcom/cal.com/commit/02b4d70831c573ccacfc8f8778e9fecf26b2e123
tried to write better cleaner functions in this commit